### DISCUSSION TITLE
[Help] Cant get a template working

### DISCUSSION THREAD

**Discussion by elpekenin** [mitsuhiko/minijinja#204] [Q&A] [answered]
1st of all, thx for your work and sorry for posting this Rust-noob question instead of some interesting idea or anything useful.

I may have missed something, but I couldn't find valuable information on docs, previous issues nor looking through GitHub to see others' code... Neither a telegram/discord/whatever chat where to ask, hence doing it here.

Im trying to write a template which receives 2 `Vec`, each one made up of a different custom type, and does something like this:
```jinja
{% for value in vec_a %}
    {% set element = vec_b.iter().find(|x| x.attr == value).unwrap(); %}
        {{ element }} 
    {% endwith %}
{% endfor %}
```

However, apparently i cant use a closure since `|` is used for filters, but i can't neither get a function or filter working, after some testing i think it has to do with `vec_b` containing a custom type doesn't fulfill a trait.

Do i need to make some `impl` for my enum, or how should i proceed?
Perhaps i could change the way in which my data is represented in the DB and/or Rust but that would require quite some work and i think my current model is ok-ish

**Comment by mitsuhiko** [comment]
MiniJinja does not have closures or anything like that, it uses the Jinja2 expression language. However the filter language is not as string yet as Jinja2 where you could express it as such:

```jinja
{% for value in vec_a %}
    {% set element = vec_b|selectattr("attr", "==", value)|first %}
    {{ element }}
{% endfor %}
```

I have been considering adding the operator filters, but it hasn't come up yet. In the interim you probably want to add a custom filter or a manual loop.

**Reply by mitsuhiko** [reply]
The missing filters will be added in the next version: https://github.com/mitsuhiko/minijinja/pull/205

**Comment by mitsuhiko** [comment, marked as answer]
With MiniJinja 0.30.3 you can express it like this:

```jinja
{% for value in vec_a %}
    {% set element = vec_b|selectattr("attr", "==", value)|first %}
    {{ element }}
{% endfor %}
```

**Reply by elpekenin** [reply]
Thank you very much for the quick addition :)
